usb/ehci/ehci-hub.c

```diff
@@ -170,23 +233,87 @@ static int ehci_clearPortFeature(usb_dev_t *hub, int port, uint16_t wValue)
 }
 
 
-static int ehci_getDesc(usb_dev_t *hub, char *buf, size_t len)
+static int ehci_getStringDesc(usb_dev_t *hub, int index, char *buf, size_t size)
 {
-	hcd_t *hcd = hub->hcd;
-	usb_hub_desc_t *desc;
+	usb_string_desc_t *desc;
+	size_t len = 0;
+	const char *src;
+	int i;
+
+	switch (index) {
+		case 0:
+			/* LangId string */
+			len = 4;
+			src = ehci_descs.langID;
+			break;
+		case 1:
+			/* Product string */
+			len = strlen(ehci_descs.product) * 2 + 3;
+			src = ehci_descs.product;
+			break;
+		case 2:
+			/* Manufacturer string */
+			len = strlen(ehci_descs.manufacturer) * 2 + 3;
+			src = ehci_descs.manufacturer;
+			break;
+		default:
+			return 0;
+	}
 
-	if (len < 9)
-		return -1;
+	if (size < len)
+		return len;
```

Comment:
At the end of this function you return the value `len`.
Is it correct to also return the `len` value here?

Reply:
It's not correct. Thanks.